fews_3di/simulation.py

```diff
@@ -166,14 +168,29 @@ def run(self):
         else:
             logger.info("Saved state not enabled in the configuration, skipping.")
 
-        rain_file = self.settings.base_dir / "input" / "precipitation.nc"
-        if rain_file.exists():
-            rain_raster_netcdf = utils.write_netcdf_with_time_indexes(
-                rain_file, self.settings
-            )
-            self._add_rain(rain_raster_netcdf)
-        else:
-            logger.info("No rain file found at %s, skipping.", rain_file)
+        if self.settings.rain_type == "constant":
+            self._add_constant_rain()
+
+        # deze functie bestaat nog niet (nog niet in api ingebouwd)
+        # elif rain_type == 'design':
+        # self._add_design_rain()
+
+        elif self.settings.rain_type == "radar":
+            self._add_radar_rain()
+
+        elif self.settings.rain_type == "custom":
+            if self.settings.rain_input == "rain_netcdf":
+                rain_netcdf = self.settings.base_dir / "input" / "precipitation.nc"
+                rain_raster_netcdf = utils.write_netcdf_with_time_indexes(
+                    rain_netcdf, self.settings
+                )
+                self._add_netcdf_rain(rain_raster_netcdf)
+            if self.settings.rain_input == "rain_csv":
+                rain_csv = self.settings.base_dir / "input" / "rain.csv"
+                rain = utils.rain_csv_timeseries(rain_csv, self.settings)
+                self._add_csv_rain(rain)
+            else:
+                logger.info("No custom rain file found, skipping.")
```

Comment:
Deze info melding klopt niet. De "else" gaat er alleen maar om of de setting "rain_csv" is.
Als de csv file niet bestaat knalt 'ie er gewoon uit met een exception. Tenminset, zo lijkt het.